backend/regalloc/regalloc_stack_slots.ml

```diff
@@ -10,29 +10,32 @@ type slot = int
 
 type t =
   { stack_slots : int Reg.Tbl.t;
-    num_stack_slots : int array
+    num_stack_slots : int Stack_class.Tbl.t
   }
 
 let[@inline] make () =
   let stack_slots = Reg.Tbl.create 128 in
-  let num_stack_slots = Array.make Proc.num_stack_slot_classes 0 in
+  let num_stack_slots = Stack_class.Tbl.make 0 in
   { stack_slots; num_stack_slots }
 
 let iter t ~f = Reg.Tbl.iter f t.stack_slots
 
-let[@inline] size_for_all_stack_classes t =
-  Array.fold_left t.num_stack_slots ~f:( + ) ~init:0
+let[@inline] total_number_of_slots t =
+  Stack_class.Tbl.fold t.num_stack_slots ~init:0 ~f:(fun _stack_class num acc ->
+      acc + num)
 
 let[@inline] get_and_incr t ~stack_class =
-  let res = t.num_stack_slots.(stack_class) in
-  t.num_stack_slots.(stack_class) <- succ res;
+  let res = Stack_class.Tbl.find t.num_stack_slots stack_class in
```

Comment:
`Tbl.update` instead?

Reply:
Unfortunately, `update` returns `unit`,
so we will have to do a `find` anyway.